algorithms/prims-algorithm/problems/cowconnection/cowconnection.cpp

```diff
@@ -1,35 +1,134 @@
+#include<queue>
 #include<unordered_map>
-#include<unordered_set>
 #include<limits>
+#include<functional>
+#include<iostream>
+#include<algorithm>
+#include<set>
 using namespace std;
 
+
+//#ifdef hackpackpp
+//Taken from the section on graphs
 typedef unordered_map<int,int> umii;
-typedef unordered_map< int , umii > umiumii;
-//The graph class uses c++11 extensions; use map to get O(log N) time with c++98
+typedef umii::const_iterator edge_iter;
```

Comment:
Something condensed should make every effort to make itself easily understandable, and this naming scheme does quite the opposite. I really think it should be changed.